sycl/doc/extensions/experimental/sycl_ext_intel_fp_control.asciidoc

```diff
@@ -0,0 +1,176 @@
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 7 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties]
+* link:../experimental/sycl_ext_oneapi_kernel_properties.asciidoc[
+  sycl_ext_oneapi_kernel_properties]
+
+== Status
+
+This is an experimental extension specification, intended to provide early access
+to features and gather community feedback. Interfaces defined in this specification
+are implemented in {dpcpp}, but they are not finalized and may change incompatibly in
+future versions of {dpcpp} without prior notice. **Shipping software products should not
+rely on APIs defined in this specification.**
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices and
+only when using the Level Zero backend or OpenCL backend. Also currently Intel GPU
+driver supports this extension only for Vector Backend, so feature is supported only
+for ESIMD.
+Attempting to use this extension in kernels that run on other devices or
+backends may result in undefined behavior.  Be aware that the compiler
+is not able to issue a diagnostic to warn you if this happens.
+
+== Overview
+
+Intel GPUs allow to control kernel behavior in floating point operations. This extension adds the kernel property `fp_control` which provides a way to specify the rounding mode, denorm mode and single precision floating point mode for floating point operations in a SYCL kernel.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FP_CONTROL` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Properties
+
+|===
+|Property|Description
+
+|`fp_control`
+|The `fp_control` kernel property provides a way to specify the rounding mode, denorm mode and single precision floating point mode for a kernel.
+|===
+
+The values that the property can be set to are a subset of the following that can be OR'ed together:
+
+[%header,cols="1,5"]
+|===
+|Mode | Description
+| rte | Round to nearest or even
+| rtp | Round towards positive infinity
+  rtn | Round towards negative infinity
+  rtz | Round towards zero
```

Comment:
I'm not very fond of these names.  I think it would be very easy to forget/misremember the explanation here, and I could see myself frequently assuming that `rtn` means "round to nearest" instead of "round to negative infinity".

What if we use names that align with C++'s `fenv` controls, instead? See [here](https://en.cppreference.com/w/cpp/numeric/fenv).  That would give us:

- `round_to_nearest`: Round to nearest or even
- `round_upward`: Round towards positive infinity
- `round_downward`: Round towards negative infinity
- `round_toward_zero`: Round towards zero

Reply:
Sounds good, thank you, will rename.

Reply:
Renamed.